### PR TITLE
feature: add ante handler to block non validators from sending restri…

### PR DESCRIPTION
**Pull request by LaurentMontBlanc**

### CODE REVIEW

**Review by eckelj — COMMENTED** [review]
I suggest to add to the list of validator only commands:
* NotarizeLiqiuidAsset is also just issued by the validators
* RegisterTrustAnchor as long as we don't have a 3rdParty OEM vendor certification system worked out in detail.

File: app/ante/check_validator_decorator.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+func NewCheckValidatorDecorator(sk StakingKeeper) CheckValidatorDecorator {
+	return CheckValidatorDecorator{sk: sk}
+}
+
+func (cv CheckValidatorDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (_ sdk.Context, err error) {
+	if simulate || ctx.BlockHeight() == 0 {
+		return next(ctx, tx, simulate)
+	}
+
+	for _, msg := range tx.GetMsgs() {
```
Comment (jmastr) [inline app/ante/check_validator_decorator.go:22]: could this be re-written to?
```
        for _, msg := range tx.GetMsgs() {
                switch sdk.MsgTypeURL(msg) {
                case "/planetmintgo.dao.MsgInitPop":
                        fallthrough
                case "/planetmintgo.dao.MsgDistributionRequest":
                        fallthrough
                case "/planetmintgo.dao.MsgDistributionResult":
                        fallthrough
                case "/planetmintgo.dao.MsgReissueRDDLProposal":
                        fallthrough
                case "/planetmintgo.dao.MsgReissueRDDLResult":
                        ctx, err = cv.handleMsg(ctx, msg)
                default:
                        continue
                }
        }
```

`cv.handleMsg()` is for all message types the same, so type assertion is unnecessary in my opinion

Comment (jmastr) [reply app/ante/check_validator_decorator.go:22]: https://golangbyexample.com/fallthrough-keyword-golang/

File: tests/e2e/machine/rest.go [hunk 2]
```diff
@@ -64,10 +64,10 @@ func (s *RestE2ETestSuite) TestAttestMachineREST() {
 	// Register TA
 	ta := sample.TrustAnchor(pubKey)
 	taMsg := machinetypes.MsgRegisterTrustAnchor{
-		Creator:     addr.String(),
+		Creator:     val.Address.String(),
```
Comment (jmastr) [inline tests/e2e/machine/rest.go:67]: before introducing the `lib` here it used to be `addr.String()` too...

**Review by jmastr — APPROVED** [review]

File: tests/e2e/machine/rest.go (hunk 2 above)
Comment (LaurentMontBlanc) [reply tests/e2e/machine/rest.go:67]: true, but with this change trust anchor registration is permitted only for validators until we have a process for registering 3rd party OEMs

**Pull request merged**